base/number.jl

```diff
@@ -168,10 +168,23 @@ abs(x::Real) = ifelse(signbit(x), -x, x)
 
 Squared absolute value of `x`.
 
+For complex numbers this is faster than `abs(x)^2`.
+
+See also [`abs`](@ref), [`conj`](@ref), [`real`](@ref).
+
 # Examples
 ```jldoctest
 julia> abs2(-3)
 9
+
+julia> abs2(3.0 + 4.0im)
+25.0
+
+julia> sum(abs2, [1+2im, 3+4im])
```

Comment:
What does this example show more than the one above? That you can give a function to `sum`? But that is true for all functions so cannot really be added in all those cases.

Reply:
Adding this to all would be madness, sure. But the proposal is more like trying to include one neat thing you might not have known, somewhat related to the function. This is `norm(x::Vector)^2`, which on paper has the same notation as `abs2(x::Number)`.

Reply:
`sum(abs2, a)` is a common idiom (at least for me), doesn't seem out of line to include it here. (in fact it would also make sense to have a `norm2` function doing that)

Reply:
Should it have a comment?
```suggestion
julia> sum(abs2, [1+2im, 3+4im])  # norm(x)^2
```